src/nvim/buffer.c

```diff
@@ -444,6 +444,9 @@ close_buffer (
     else
       buf->b_next->b_prev = buf->b_prev;
     free_buffer(buf);
+    if (!autocmd_busy && curbuf == buf) {
+      curbuf = firstbuf;
+    }
```

Comment:
@justinmk , I placed the check after `free_buffer()`, because somewhere in the code path `curbuf` could be used (probably not). It should be safe, because `buf` is freed only with the last statement in `free_buffer()`.

Bram made a patch, https://github.com/vim/vim/commit/62ef797496c6243d111c596a592a8ef8c1d1e710, where he placed the check "after" `free_buffer()`. One difference here is, `curbuf` is also changed while autocommands are active (and the buffer is not freed yet).